src/linux/component.rs

```diff
@@ -80,13 +80,21 @@ fn append_files(components: &mut Vec<Component>, folder: &Path) {
                     _ => {}
                 }
             }
-            if let (Some(_), Some(found_input)) = (found_label, found_input) {
+
+            if let Some(found_input) = found_input {
                 let mut p_label = folder.to_path_buf();
                 let mut p_input = folder.to_path_buf();
                 let mut p_crit = folder.to_path_buf();
                 let mut p_max = folder.to_path_buf();
 
-                p_label.push(&format!("temp{}_label", key));
+                // Disk have no labe why Don't know.
+                // So we use model name instead
+                let fragment_label = match found_label {
+                    Some(_) => format!("temp{}_label", key),
+                    None => "device/model".to_string(),
```

Comment:
If we're sure it's only impacting dis, we could prepend this label by `Disk ` or something like that. What do you think?

Reply:
The main thing is that I change the file to be opened. Disk drive don't have `temp*_label` files so I use relative link `device` to open the `model` name.

Here an example on my computer:
```
ls -alh /sys/class/hwmon/hwmon7/
total 0
drwxr-xr-x 3 root root    0 Aug 28 23:28 .
drwxr-xr-x 3 root root    0 Aug 28 23:28 ..
lrwxrwxrwx 1 root root    0 Aug 28 23:30 device -> ../../../0:0:0:0
-r--r--r-- 1 root root 4.0K Aug 28 23:30 name
drwxr-xr-x 2 root root    0 Aug 28 23:33 power
lrwxrwxrwx 1 root root    0 Aug 28 23:33 subsystem -> ../../../../../../../../../class/hwmon
-r--r--r-- 1 root root 4.0K Aug 28 23:30 temp1_input
-rw-r--r-- 1 root root 4.0K Aug 28 23:28 uevent
```

It's not a problem related with our `HashMap` and the keys, it's a problem in `hwmon` not exposing `temp*_label` for disk (and other devices!).

Reply:
And by the way `name` is not useful it expose the probe name  here `drivetemp`

Reply:
It may be worth it to test with an other probe writing to hwmon (there is some?)